### DISCUSSION TITLE
A framework to support spinning test environments to be used for functional testing

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#20726] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

As far as I understand, functional testing are executed by contributors manually while following the documentation on how to setup the environment.
Instead, a framework can be used to setup an environment for testing, populated with data using DaC (Data as Code), tested with TaC (Tests as Code).
One option is to spin containers for the various Git Forges (where applicable)  or use a dedicated org for setting up repos under tests.

**Comment by viceice** [comment]
i think we should use samples/ reproductions from GitHub as test repos ( copy / clone to target platform) instead of creating new fixtures inside this repo. so we don't blow up this repo with e2e test data.

**Reply by lmilbaum** [reply]
Thanks Michael for suggesting it. I can initiate a write up for the high level architecture if that is something that can help pushing this forward. WDYT?

**Reply by viceice** [reply]
sounds good

@rarkins WDYT?

**Comment by lmilbaum** [comment]
[WIP document](https://docs.google.com/document/d/1Bn0P3Xdm_5J0KrCAMykEALyNvbt1IXT8Gj8CagTG6as/edit?usp=sharing)
